src/v/cloud_storage/partition_manifest.cc

```diff
@@ -230,11 +232,58 @@ model::initial_revision_id partition_manifest::get_revision_id() const {
 
 remote_segment_path partition_manifest::generate_segment_path(
   const partition_manifest::key& key, const segment_meta& meta) const {
-    auto name = generate_segment_name(key.base_offset, key.term);
-    return generate_remote_segment_path(
+    auto name = cloud_storage::generate_local_segment_name(
+      key.base_offset, key.term);
+    return cloud_storage::generate_remote_segment_path(
       _ntp, meta.ntp_revision, name, meta.archiver_term);
 }
 
+segment_name partition_manifest::generate_remote_segment_name(
+  const partition_manifest::key& k, const partition_manifest::value& val) {
+    switch (val.sname_format) {
+    case segment_name_format::v1:
+        if (val.archiver_term != model::term_id{}) {
+            // Use format ".../offset-term-v1.log.archiver-term"
+            // The metadata is still v1, but we have an archiver term
+            return segment_name(ssx::sformat(
+              "{}-{}-v1.log.{}", k.base_offset, k.term, val.archiver_term));
+        } else {
+            // Use old format ".../offset-term-v1.log"
+            // The manifest is v21.9 or earlier
+            return segment_name(
+              ssx::sformat("{}-{}-v1.log", k.base_offset, k.term));
+        }
+    case segment_name_format::v2:
+        // Use new stlyle format ".../base-committed-term-v1.log.archiver-term"
+        return segment_name(ssx::sformat(
+          "{}-{}-{}-v1.log.{}",
+          k.base_offset,
+          val.committed_offset,
+          k.term,
+          val.archiver_term));
```

Comment:
I think this archiver term gets appended twice, once here and then once in `cloud_storage::generate_remote_segment_path`, for example when calling `partition_manifest::generate_remote_segment_path` with a key and meta for v2.

Reply:
fixed